src/app/fdctl/run/tiles/fd_net.c

```diff
@@ -243,29 +233,8 @@ after_frag( void *             _ctx,
 
   fd_aio_pkt_info_t aio_buf = { .buf = ctx->frame, .buf_sz = (ushort)*opt_sz };
   if( FD_UNLIKELY( route_loopback( ctx->src_ip_addr, *opt_sig ) ) ) {
-    /* Because we are skipping part of the kernel networking stack, we
-       need to reformat the packet so it will get accepted onto the
-       loopback device.  A few things are needed,
-       
-        (1) The src and dst ip addresses both need to be zero. Not
-            entirely sure why, but the kernel will silently discard
-            packets that arrive onto lo without this, probably it
-            sets them to zero as part of a sanity check in the code
-            that runs before the XDP splice point.
-            
-        (2) The src mac address can be anything, but the dst mac
-            address must be zero.  Similar story to the above, not
-            entirely sure why. */
-    eth_ip_udp_t * hdr = (eth_ip_udp_t *)ctx->frame;
     if( FD_UNLIKELY( *opt_sz < sizeof( eth_ip_udp_t ) ) )
-      FD_LOG_ERR(( "loopback packet too small %lu", *opt_sz ));
-
-    fd_memset( hdr->eth->dst, 0, 6UL );
-    fd_memset( hdr->ip4->daddr_c, 0, 4UL );
-    fd_memset( hdr->ip4->saddr_c, 0, 4UL );
-    hdr->ip4->check = 0U;
-    /* TODO: Do we need to update check here? */
-    hdr->ip4->check = fd_ip4_hdr_check( ( fd_ip4_hdr_t const *) FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4 ) );
+      FD_LOG_WARNING(( "loopback packet too small %lu", *opt_sz ));
```

Comment:
Why change this to WARNING? Should never happen.

Reply:
I ended up removing this check - I had changed it to warning so we wouldn't crash at runtime if sending a small packet due to a bug. LMK if you want it added back.